### ISSUE TITLE
Renderer OpenGL 4 failed to initialize properly

### ISSUE THREAD

**Issue by DaxterJunior** [grimfang4/sdl-gpu#109]
Hi, I started to use SDL_gpu, but whenever I want to load an image I get these errors:

`Failed to create GL context`
`Renderer OpenGL 4 failed to initialize properly`
`Failed to load from rwops: PNG not supported: 1/2/4/8-bit only`
`Failed to load image data.`

With SDL2 this never happened, so I don't know if this is a bug or if I'm doing something wrong.

**Comment by grimfang4** [comment]
It's okay for the GL 4 renderer to fail, just as long as you do eventually get a renderer to succeed (likely GL 3).  SDL_gpu checks a sequence of renderers as a fall-back.

Can you send me the image you're trying to load?  The error message may be right (format unsupported by stb-image) or it might be masking some other issue.  Did you use SDL_image to load it with SDL2?  If nothing else, you can still use SDL_image's IMG_Load() and then SDL_gpu's GPU_CopyImageFromSurface().

**Comment by DaxterJunior** [comment]
Ok, I fixed as you said, using IMG_Load()  and then GPU_CopyImageFromSurface. It works, thanks.

**Comment by grimfang4** [comment]
Is there a chance you can pass along an image that fails like this?  I'd love to fix it or else issue a clearer error message or improve the documentation.

**Comment by DaxterJunior** [comment]
Yes, I forgot it. Here is a link: [imgur](https://imgur.com/1j90A3G)

**Comment by grimfang4** [comment]
pngcheck says the file in question is a 64-bit PNG.  SDL_gpu passes along the error message from stb-image (the image loader used by SDL_gpu internally), which says it supports 1, 2, 4, and 8-bit channels. 
 It's not totally clear whether stb-image is intended to support 64-bit PNGs.

Sometime, I'll see if an update to stb-image fixes it.

Your options for now are to use a different image loader, like SDL_image (which uses the more comprehensive libpng), or convert the file to a lower bit depth (e.g. load and re-export using an image editor).
